### PR TITLE
feat: read lyrics from ID3v2 tag

### PR DESCRIPTION
**Pull request by kt286**
read lyrics from  ID3v2 tag when the *.lrc file not found in media file root dir

### CODE REVIEW

**Comment by Sunnshino** [comment]
这是来自QQ邮箱的假期自动回复邮件。
 
您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。

**Comment by iamhyc** [comment]
@babyfengfjx @Sunnshino Could someone kindly help review this PR?

 It seems like the maintainer "Wang Yu" is not on Github.

**Review by wyu0701 — COMMENTED** [review]
The 'byteArray' is not used, it is recommended to delete it

**Comment by kt286** [comment]
> The 'byteArray' is not used, it is recommended to delete it

modified

**Comment by kt286** [comment]
@wangyu0701 test music: https://github.com/linuxdeepin/deepin-music/files/7923703/-.Lao.zip

**Comment by lzwind** [comment]
This is a new function. We are evaluating it. It will take some time. Please wait patiently. Thank you!

File: src/music-player/presenter/dboperate.cpp [hunk 1]
```diff
@@ -217,6 +217,25 @@ void DBOperate::slotCreatCoverImg(const QList<MediaMeta> &metas)
...
+        if (m_needStop) {
+            break;
+        }
+        if (m_needSleep) {
+            QThread::msleep(SLEEPTIME);
+            m_needSleep = false;
+        }
+
+        //没有加载过的文件才去解析数据
+        QFileInfo lyricInfo(Global::cacheDir() + "/lyrics/" + meta.hash + ".lrc");
```
Comment (zccrs) [inline src/music-player/presenter/dboperate.cpp:232]: 这个路径的组合我看不止这一个地方在用，建议封装成一个 inline static 的功能函数，在用到的地方换成函数调用，避免以后有相关逻辑需要改动的时候可能漏掉了改另外相同的地方。

File: src/music-player/widget/lrc/musiclyricwidget.cpp [hunk 2]
```diff
@@ -227,13 +227,22 @@ void MusicLyricWidget::onMusicPlayed(MediaMeta meta)
...
     QFile file(lrcPath);
     if (!file.exists()) {
-        m_nolyric->show();
-        m_lyricview->hide();
+        //读取根目录中歌词文件后读取媒体文件内嵌的歌词
+        MediaMeta meta = Player::getInstance()->getActiveMeta();
+        QFileInfo lyricInfo(Global::cacheDir() + "/lyrics/" + meta.hash + ".lrc");
+        if (!lyricInfo.exists()) {
+            m_nolyric->show();
+            m_lyricview->hide();
```
Comment (zccrs) [inline src/music-player/widget/lrc/musiclyricwidget.cpp:235]: 最好是先hide再show另一个，虽然实际感官上看不出区别，不过逻辑上这样会更合适

**Review by zccrs — CHANGES_REQUESTED** [review]

File: src/music-player/presenter/dboperate.cpp (hunk 1 above)
Comment (kt286) [reply src/music-player/presenter/dboperate.cpp:232]: 分散在3个文件。不知道该定义到哪里

**Pull request closed without merging**